internal/types/utils.go

```diff
@@ -325,3 +325,22 @@ func CheckListKeyMatch(k string, list []string) bool {
 	}
 	return false
 }
+
+// Add the password and encrypted_password to existing map[string]attr.Type
+func AddPasswordToMapStringAttrType(mapStringAttrType map[string]attr.Type) map[string]attr.Type {
```

Comment:
You could combine these into one generic method. Just make the name of the attribute a parameter